### ISSUE TITLE
unhandled promise rejection: Cannot read property 'getEntries' of null

### ISSUE THREAD

**Issue by DutchmanNL** [ioBroker/ioBroker.zigbee#642]
```

2020-05-22 09:09:04.942 - error: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
--
2020-05-22 09:09:04.942 - error: zigbee.0 (28760) unhandled promise rejection: Cannot read property 'getEntries' of null
2020-05-22 09:09:04.943 - error: zigbee.0 (28760) TypeError: Cannot read property 'getEntries' of null
at Function.loadFromDatabaseIfNecessary (/opt/iobroker/node_modules/zigbee-herdsman/dist/controller/model/device.js:200:55)
at Function.byIeeeAddr (/opt/iobroker/node_modules/zigbee-herdsman/dist/controller/model/device.js:208:16)
at Controller.getDeviceByIeeeAddr (/opt/iobroker/node_modules/zigbee-herdsman/dist/controller/controller.js:235:31)
at ZigbeeController.resolveEntity (/opt/iobroker/node_modules/iobroker.zigbee/lib/zigbeecontroller.js:232:52)
at Zigbee.publishFromState (/opt/iobroker/node_modules/iobroker.zigbee/main.js:196:48)
at StatesController.emit (events.js:198:13)
at StatesController.publishFromState (/opt/iobroker/node_modules/iobroker.zigbee/lib/statescontroller.js:142:14)
at process._tickCallback (internal/process/next_tick.js:68:7)
2020-05-22 09:09:04.943 - info: zigbee.0 (28760) cleaned everything up...
2020-05-22 09:09:04.944 - info: zigbee.0 (28760) Zigbee: disabling joining new devices.
2020-05-22 09:09:04.945 - error: zigbee.0 (28760) Failed to stop zigbee (TypeError: Cannot read property 'getEntries' of null
at Function.loadFromDatabaseIfNecessary (/opt/iobroker/node_modules/zigbee-herdsman/dist/controller/model/group.js:64:55)
at Function.all (/opt/iobroker/node_modules/zigbee-herdsman/dist/controller/model/group.js:76:15)
at Controller. (/opt/iobroker/node_modules/zigbee-herdsman/dist/controller/controller.js:178:49)
at Generator.next ()
at /opt/iobroker/node_modules/zigbee-herdsman/dist/controller/controller.js:8:71
at new Promise ()
at __awaiter (/opt/iobroker/node_modules/zigbee-herdsman/dist/controller/controller.js:4:12)
at Controller.stop (/opt/iobroker/node_modules/zigbee-herdsman/dist/controller/controller.js:174:16)
at ZigbeeController.stop (/opt/iobroker/node_modules/iobroker.zigbee/lib/zigbeecontroller.js:307:33)
at process._tickCallback (internal/process/next_tick.js:68:7))
2020-05-22 09:09:04.987 - info: zigbee.0 (28760) terminating
2020-05-22 09:09:04.988 - info: zigbee.0 (28760) Terminated (NO_ERROR): Without reason
2020-05-22 09:09:05.503 - info: host.ioBroker instance system.adapter.zigbee.0 terminated with code 0 (NO_ERROR)
2020-05-22 09:09:05.503 - info: host.ioBroker Restart adapter system.adapter.zigbee.0 because enabled
2020-05-22 09:09:35.519 - info: host.ioBroker instance system.adapter.zigbee.0 started with pid 25459
2020-05-22 09:09:36.260 - info: zigbee.0 (25459) starting. Version 1.1.1 in /opt/iobroker/node_modules/iobroker.zigbee, node: v10.20.1, js-controller: 3.1.4
2020-05-22 09:09:36.272 - info: zigbee.0 (25459) Starting Zigbee...
2020-05-22 09:09:43.420 - error: zigbee.0 (25459) Failed to start Zigbee
2020-05-22 09:09:43.420 - error: zigbee.0 (25459) Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
at ZStackAdapter. (/opt/iobroker/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:71:23)
at Generator.throw ()
at rejected (/opt/iobroker/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:6:65)
2020-05-22 09:09:53.420 - info: zigbee.0 (25459) Try to reconnect. 1 attempts left
2020-05-22 09:09:53.421 - info: zigbee.0 (25459) Starting Zigbee...
2020-05-22 09:09:53.529 - error: zigbee.0 (25459) Failed to start Zigbee
2020-05-22 09:09:53.529 - error: zigbee.0 (25459) Error: Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'
at Znp. (/opt/iobroker/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:113:32)
at Generator.next ()
at /opt/iobroker/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:8:71
at new Promise ()
at __awaiter (/opt/iobroker/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:4:12)
at SerialPort.serialPort.open (/opt/iobroker/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:111:49)
at SerialPort._error (/opt/iobroker/node_modules/@serialport/stream/lib/index.js:198:14)
at binding.open.then.err (/opt/iobroker/node_modules/@serialport/stream/lib/index.js:242:12)


```

**Comment by arteck** [comment]
2020-05-22 09:09:53.529 - error: zigbee.0 (25459) Error: Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'

**Comment by Matze1985** [comment]
`zigbee.0 | 2020-05-25 12:06:32.406 | error | at /opt/iobroker/node_modules/zigbee-herdsman/node_modules/@serialport/stream/lib/index.js:242:12
-- | -- | -- | --
zigbee.0 | 2020-05-25 12:06:32.406 | error | at SerialPort._error (/opt/iobroker/node_modules/zigbee-herdsman/node_modules/@serialport/stream/lib/index.js:198:14)
zigbee.0 | 2020-05-25 12:06:32.406 | error | at SerialPort.<anonymous> (/opt/iobroker/node_modules/zigbee-herdsman/dist/adapter/deconz/driver/driver.js:89:45)
zigbee.0 | 2020-05-25 12:06:32.406 | error | at __awaiter (/opt/iobroker/node_modules/zigbee-herdsman/dist/adapter/deconz/driver/driver.js:4:12)
zigbee.0 | 2020-05-25 12:06:32.406 | error | at new Promise (<anonymous>)
zigbee.0 | 2020-05-25 12:06:32.406 | error | at /opt/iobroker/node_modules/zigbee-herdsman/dist/adapter/deconz/driver/driver.js:8:71
zigbee.0 | 2020-05-25 12:06:32.406 | error | at Generator.next (<anonymous>)
zigbee.0 | 2020-05-25 12:06:32.406 | error | at Driver.<anonymous> (/opt/iobroker/node_modules/zigbee-herdsman/dist/adapter/deconz/driver/driver.js:91:28)
zigbee.0 | 2020-05-25 12:06:32.406 | error | (2907) Error: Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'
zigbee.0 | 2020-05-25 12:06:32.405 | error | (2907) Failed to start Zigbee
zigbee.0 | 2020-05-25 12:06:32.241 | info | (2907) Starting Zigbee...
zigbee.0 | 2020-05-25 12:06:32.240 | info | (2907) Try to reconnect. 1 attempts left`

Same problems

**Comment by waynemuh** [comment]
Unplug USB device and plug in again helps

**Comment by incase** [comment]
I wonder if I could re-open this issue. I'm seeing the same problem and unplugging and re-plugging helps sometimes but far from every time.

**Comment by incase** [comment]
Thanks for re-opening.
So what I found out so far:
zigbeecontroller.js did call herdsman start()
However, I suspect that said start didn't complete by the time the getDeviceByIeeeAddr is called.

See also https://github.com/Koenkk/zigbee-herdsman/issues/280 where I asked for help resolving this issue.

**Comment by asgothian** [comment]
In order to identify what is happening, we need a complete log of the issue, in an ideal case in a way that it can be replicated. From what i read above:

- If you unplug and replug the zigbee Stick, the Zigbeeadapter will start and turn green.
- You do something (not sure what yet) and the adapter crashes.
- Repeated start Attempts all end because the serial port cannot be opened.

If this is correct, we need to know exactly what happens when the adapter crashes as well as the log for this case (complete log, i.e. downloaded from the system, not just a copy or screenshot from admin.

A.

**Comment by incase** [comment]
We switched to a different setup, so I unfortunately can't help getting debug information here anymore. Sorry about that.
As for "what happens when the adapter crashes": I have absolutely no idea what exactly might have caused that. From what I gathered from various sources, it might have been an issue with the device we used before running out of memory, but I can't confirm or deny that, to be honest. We do have a fair number of zigbee devices strewn all over the house, so routing sometimes got complex.
